pod/annotations.go

```diff
@@ -606,5 +607,22 @@ func IsPlatformSidecarContainer(name string, pod *corev1.Pod) bool {
 
 // SidecarAnnotation forms an annotation key referencing a particular sidecar.
 func SidecarAnnotation(sidecarName, suffix string) string {
-	return fmt.Sprintf("%s.%s/%s", AnnotationKeyPrefixSidecars, sidecarName, suffix)
+	return fmt.Sprintf("%s.%s/%s", sidecarName, AnnotationKeySuffixSidecars, suffix)
+}
+
+// LegacySidecarAnnotation forms an annotation key referencing a particular
+// sidecar in the old format.
+// TODO(aaronl): Remove this once we've fully transitioned to the new format.
+func LegacySidecarAnnotation(sidecarName, suffix string) string {
+	return fmt.Sprintf("%s.%s/%s", AnnotationKeyPrefixSidecarsLegacy, sidecarName, suffix)
+}
+
+// SidecarAnnotations returns both current and legacy forms of an annotation key
+// referencing a particular sidecar.
+// TODO(aaronl): Remove this once we've fully transitioned to the new format.
+func SidecarAnnotations(sidecarName, suffix string) []string {
```

Comment:
Are you sure you don't even want one higher level function that abstracts on top of this, that actually strips the annotations and returns `[]PlatformSidecar{}` ?

Reply:
I think it's a good idea. Let's consider it as a followup. It would take a bit of refactoring in the `platform-sidecar-mutator` code base (currently it looks up annotations on demand) and I don't want to block your progress.